--- winesync.c
+++ winesync.c
@@ -10,7 +10,7 @@
 #include <linux/module.h>
 #include <linux/slab.h>
 #include <linux/xarray.h>
-#include <uapi/linux/winesync.h>
+#include "winesync.h"
 
 #define WINESYNC_NAME	"winesync"
 
@@ -1187,7 +1187,7 @@
 };
 
 static struct miscdevice winesync_misc = {
-	.minor		= WINESYNC_MINOR,
+	.minor		= MISC_DYNAMIC_MINOR,
 	.name		= WINESYNC_NAME,
 	.fops		= &winesync_fops,
 };
@@ -1209,4 +1209,3 @@
 MODULE_DESCRIPTION("Kernel driver for Wine synchronization primitives");
 MODULE_LICENSE("GPL");
 MODULE_ALIAS("devname:" WINESYNC_NAME);
-MODULE_ALIAS_MISCDEV(WINESYNC_MINOR);
